utam-compiler/src/main/java/utam/compiler/grammar/UtamMethodActionWaitFor.java

```diff
@@ -186,18 +187,34 @@ Operand getOperand() {
   static class UtamMethodActionWaitForElement extends UtamMethodActionWaitFor {
 
     private final List<UtamArgument> args;
+    private final boolean isNoArgsAllowed;
+    private final String elementName;
+    private final String elementGetterName;
 
-    UtamMethodActionWaitForElement(String elementName) {
+    UtamMethodActionWaitForElement(String elementName, boolean isNoArgsAllowed, boolean isPublicElement) {
       super(null, "waitFor", null, null, false);
       UtamMethodAction getter = new UtamMethodActionGetter(elementName, null, null, null, null,
           false);
       UtamArgument argument = new UtamArgumentPredicate(getter);
       this.args = Collections.singletonList(argument);
+      this.isNoArgsAllowed = isNoArgsAllowed;
+      this.elementName = elementName;
+      this.elementGetterName = getElementGetterName(elementName, isPublicElement);
+    }
+    private static final String getElementGetterName(String elementName, boolean isPublicElement) {
+      //similar logic in BasicElementUnionType.getBasicTypeName?
+      return "get" + elementName.substring(0, 1).toUpperCase() + elementName.substring(1) + (isPublicElement ? "" : "Element");
     }
 
     @Override
     Statement getStatement(TranslationContext context, MethodContext methodContext,
         StatementContext statementContext) {
+      /* The validation is performed here as we do not know args at the time of construction*/
```

Comment:
nit: pls use // and lower case, we don't use /* for one line comments in this repo